src/MoBi.Presentation/UICommand/CommitSimulationSolverAndSchemaUICommand.cs

```diff
@@ -0,0 +1,30 @@
+﻿using MoBi.Core.Domain.Model;
+using MoBi.Presentation.Tasks.Interaction;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Services;
+using OSPSuite.Presentation.UICommands;
+
+namespace MoBi.Presentation.UICommand
+{
+   public class CommitSimulationSolverAndSchemaUICommand : ActiveObjectUICommand<SimulationSettings>
+   {
+      private readonly IInteractionTasksForSimulationSettings _simulationSettingsTask;
+      private readonly IMoBiContext _context;
+
+      public CommitSimulationSolverAndSchemaUICommand(
+         IInteractionTasksForSimulationSettings simulationSettingsTask,
+         IMoBiContext context,
+         IActiveSubjectRetriever activeSubjectRetriever) :
+         base(activeSubjectRetriever)
+      {
+         _simulationSettingsTask = simulationSettingsTask;
+         _context = context;
+      }
+
+      protected override void PerformExecute()
+      {
+         var clonedSimulationSettings = _context.Clone(Subject);
```

Comment:
I wonder if this is the place to do it. Just a question. Maybe the settingsTask should e the one cloning instead of the caller? not sure